### ISSUE TITLE
It is not possible to use ENV variable for log level

### ISSUE THREAD

**Issue by asp24** [symfony/monolog-bundle#220]
Due to https://github.com/symfony/monolog-bundle/blob/master/DependencyInjection/MonologExtension.php#L37

it is not possible to specify log level using '%env(LOG_LEVEL%)' string as value

```yaml
monolog:
    handlers:
        main:
            type:         'fingers_crossed'
            action_level: '%env(LOG_LEVEL)%'
            handler:      'nested'
            buffer_size:  10
        nested:
            type:  'stream'
            path:  '%kernel.logs_dir%/%kernel.environment%.log'
            level: 'debug'
        console:
            type:  'console'
```

```
PHP Fatal error:  Uncaught Symfony\Component\Debug\Exception\ContextErrorException: Warning: constant(): Couldn't find constant Monolog\Logger::%env(LOG_LEVEL)% in ./vendor/symfony/monolog-bundle/DependencyInjection/MonologExtension.php:42
```

**Comment by ababushkin** [comment]
Any update on this one?

**Comment by alterphp** [comment]
Going to be fixed at Symfony level (many other bundles configs have the same problem) with https://github.com/symfony/symfony/pull/23888

**Comment by stof** [comment]
this particular case will *not* be fixed by the Symfony change (turning the config into the constant value simply cannot be done without having the *value* of the env var)

**Comment by alterphp** [comment]
And what about an intermediate parameter as following : 
```

**Comment by alterphp** [comment]
@stof  And what about an intermediate parameter ?

```
parameters:
    monolog.main.level: '%env(MONOLOG_LOG_LEVEL)%'
monolog:
    handlers:
        main:
            type:         'fingers_crossed'
            action_level: '%monolog.main.level%'
```

Will the symfony PR fix that case ?

**Comment by stof** [comment]
Nope, because the parameters gets resolved already, so this is exactly the same case than using the env placeholder directly. In any case, the conversion of the constant name to the constant value **cannot** work when having a placeholder instead of the actual config.

The Symfony change will make things work in places performing a type validation (so places enforcing an integer will accept the placeholder if it is a `%env(int:...)%` placeholder casting to int). But it cannot make complex transformations work.

**Comment by alterphp** [comment]
Ok @stof , thanks for details.

How do you handle multiple stages log level configuration at Sensio ? Is the production value fixed in the config ?

**Comment by ClementGautier** [comment]
Hello there ! 

So what the best workaround in your opinion guys ?
I'm tempted to create an other env for debug purpose like "debug" so even in production by changing the SYMFONY_ENV variable in order to have more logs but there is probably a better solution.

**Comment by netiul** [comment]
@stof As far as I can see the problem of unable to use environment variables is the casting of the configured level to a monolog constant at Symfony's container compile time. But the monolog/monolog project already takes care of PSR-3 levels since [1.11](https://github.com/Seldaek/monolog/commit/feb19af2d42ab598e3e5e77e8f4bd51b82538acb) in all its public API's. So the conversion in the Extension seems to be redundant. If we remove it, the usage of environment variables just becomes available without breaking the usage of PSR level naming.

What do you think?

**Comment by eschultz-magix** [comment]
+1 and I wrote a small extension to resolve the variable earlier. Unfortunately, this requires to know the handler name, but as a quick hack this will work:

```php
// App/DependencyInjection/EnvVarAsMonologLevelExtension
<?php
namespace App\DependencyInjection;

use Symfony\Component\DependencyInjection\ContainerBuilder;
use Symfony\Component\DependencyInjection\Extension\Extension;
use Symfony\Component\DependencyInjection\Extension\PrependExtensionInterface;

class EnvVarAsMonologLevelExtension extends Extension implements PrependExtensionInterface
{
    public function load(array $configs, ContainerBuilder $container)
    {
        // empty
    }

    public function prepend(ContainerBuilder $container)
    {
        if ($container->hasExtension('monolog')) {
            $level = $container->resolveEnvPlaceholders('%env(ENV_NAME)%', true);

            $container->prependExtensionConfig(
                'monolog',
                array(
                    'handlers' => array(
                        'YOUR_HANDLER_NAME' => array(
                            'level' => $level,
                        ),
                    ),
                )
            );
        }
    }
}
```

```php
// App/Kernel.php
namespace App;

use App\DependencyInjection\EnvVarAsMonologLevelExtension;

class Kernel extends BaseKernel
{
    protected function build(ContainerBuilder $container)
    {
        $container->registerExtension(new EnvVarAsMonologLevelExtension());
    }
}

You also need to remove the "level" key from the monolog configuration, otherwise it will override the value set by this extension.

**Comment by chokri** [comment]
Hi,
I'm wondering if this feature is supported by newest Symfony version ">4.*", if anyone has tested this?

**Comment by stof** [comment]
> How do you handle multiple stages log level configuration at Sensio ? Is the production value fixed in the config ?

no idea. I'm not working at Sensio.

**Comment by adegrenand** [comment]
Hi,
Any news about that. It will be very useful !

**Comment by lyrixx** [comment]
Fixed via #357
